Java Heap Space Error

###### Reproduction Steps

1. Convert a java file to kotlin (in this case, PreferenceExtensions.java)
2. Commit > Extra Commit for .java > .kt renames

EDIT: Amending conversion commits also fails. Screen capture below for reference. 

###### Expected Result
1 File Committed

###### Actual Result
`0 file committed, 1 file failed to commit: [Kotlin Migration] PreferenceExtensions package: com.ichi2.preferences Running Ktlint over these files: AnkiDroid/src/main/java/com/ichi2/preferences/PreferenceExtensions.kt FAILURE: Build failed with an exception. * What went wrong: Execution failed for task ':AnkiDroid:runKtlintFormatOverMainSourceSet'. > A failure occurred while executing org.jlleitschuh.gradle.ktlint.worker.KtLintWorkAction > Java heap space * Try: > Run with --stacktrace option to get the stack trace. > Run with --info or --debug option to get more log output. > Run with --scan to get full insights. * Get more help at https://help.gradle.org BUILD FAILED in 4s Cleaning the project and retrying. > Configure project :AnkiDroid WARNING:The option setting 'android.jetifier.ignorelist=bcprov' is experimental. > Task :AnkiDroid:clean > Task :api:clean > Task :lint-rules:clean Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0. You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins. See https://docs.gradle.org/7.3.3/userguide/command_line_interface.html#sec:command_line_warnings BUILD SUCCESSFUL in 1s 3 actionable tasks: 3 executed FAILURE: Build failed with an exception. * What went wrong: Execution failed for task ':AnkiDroid:runKtlintFormatOverMainSourceSet'. > A failure occurred while executing org.jlleitschuh.gradle.ktlint.worker.KtLintWorkAction > Java heap space * Try: > Run with --stacktrace option to get the stack trace. > Run with --info or --debug option to get more log output. > Run with --scan to get full insights. * Get more help at https://help.gradle.org BUILD FAILED in 4s`


###### Research
What was tried: 
Unchecking git hooks in the commit window allows for proper execution of commits. Problem lies with commit hooks, specifically lint check. 
(to circumvent the issue, cleaning the project, restarting the IDE and rebuilding worked for the most part)

1. Settings > System Settings > Memory settings > Kotlin daemon max heap size (currently 3072 MB) -> change to 4096MB. 
1.1 Output: No change in commit failure
2. _ktlint {
  version = "0.43.0"
}_ in **build.gradle** 
2.1. Output: Commit execution not getting stopped
3. _tasks.withType<org.jlleitschuh.gradle.ktlint.tasks.BaseKtLintCheckTask> {
    workerMaxHeapSize.set("4096M")
}_ in **gradle.properties**
3.1. Output: Worked. Commit not failing anymore. (EDIT* Not working on amend failures)

NOTE: Once the commit were executed without fail , the failure could not be reproduced any longer(despite not changing the build.gradle even in different branches and even resetting the same branch to the point of failure from the reflog).   

EDIT: When trying to amend https://github.com/ankidroid/Anki-Android/pull/10249, same commit failure occurs again. 
![HeapSpaceError_commitFailure](https://user-images.githubusercontent.com/81802035/150078330-246423c0-2e84-45e2-aff7-c89fd97fa5c7.gif)

--- a/AnkiDroid/build.gradle
+++ b/AnkiDroid/build.gradle
@@ -205,6 +205,7 @@ android {
     ndkVersion "22.0.7026061"
 
     ktlint {
+        version = "0.45.1"
         disabledRules = ["no-wildcard-imports"]
     }
 }

--- a/AnkiDroid/src/main/java/com/ichi2/anki/ReadText.kt
+++ b/AnkiDroid/src/main/java/com/ichi2/anki/ReadText.kt
@@ -142,211 +142,210 @@ object ReadText {
                 Timber.w(e, "Activity invalidated before TTS language dialog could display")
             }
         }, delayMillis.toLong())
+    }
+
+    /**
+     * Read a card side using a TTS service.
+     *
+     * @param textsToRead      Data for the TTS to read
+     * @param cardSide         Card side to be read; SoundSide.SOUNDS_QUESTION or SoundSide.SOUNDS_ANSWER.
+     * @param did              Index of the deck containing the card.
+     * @param ord              The card template ordinal.
+     */
+    fun readCardSide(textsToRead: List<TTSTag>, cardSide: SoundSide, did: Long, ord: Int) {
+        var isFirstText = true
+        var playedSound = false
+        for (textToRead in textsToRead) {
+            if (textToRead.fieldText.isEmpty()) {
+                continue
+            }
+            playedSound = playedSound or textToSpeech(
+                textToRead, did, ord, cardSide,
+                if (isFirstText) TextToSpeech.QUEUE_FLUSH else TextToSpeech.QUEUE_ADD
+            )
+            isFirstText = false
         }
+        // if we didn't play a sound, call the completion listener
+        if (!playedSound) {
+            mCompletionListener!!.onDone(cardSide)
+        }
+    }
 
-        /**
-         * Read a card side using a TTS service.
-         *
-         * @param textsToRead      Data for the TTS to read
-         * @param cardSide         Card side to be read; SoundSide.SOUNDS_QUESTION or SoundSide.SOUNDS_ANSWER.
-         * @param did              Index of the deck containing the card.
-         * @param ord              The card template ordinal.
-         */
-        fun readCardSide(textsToRead: List<TTSTag>, cardSide: SoundSide, did: Long, ord: Int) {
-            var isFirstText = true
-            var playedSound = false
-            for (textToRead in textsToRead) {
-                if (textToRead.fieldText.isEmpty()) {
-                    continue
-                }
-                playedSound = playedSound or textToSpeech(
-                    textToRead, did, ord, cardSide,
-                    if (isFirstText) TextToSpeech.QUEUE_FLUSH else TextToSpeech.QUEUE_ADD
-                )
-                isFirstText = false
-            }
-            // if we didn't play a sound, call the completion listener
-            if (!playedSound) {
-                mCompletionListener!!.onDone(cardSide)
+    /**
+     * Read the given text using an appropriate TTS voice.
+     *
+     *
+     * The voice is chosen as follows:
+     *
+     *
+     * 1. If localeCode is a non-empty string representing a locale in the format returned
+     * by Locale.toString(), and a voice matching the language of this locale (and ideally,
+     * but not necessarily, also the country and variant of the locale) is available, then this
+     * voice is used.
+     * 2. Otherwise, if the database contains a saved language for the given 'did', 'ord' and 'qa'
+     * arguments, and a TTS voice matching that language is available, then this voice is used
+     * (unless the saved language is NO_TTS, in which case the text is not read at all).
+     * 3. Otherwise, the user is asked to select a language from among those for which a voice is
+     * available.
+     *
+     * @param queueMode TextToSpeech.QUEUE_ADD or TextToSpeech.QUEUE_FLUSH.
+     * @return false if a sound was not played
+     */
+    private fun textToSpeech(tag: TTSTag, did: Long, ord: Int, qa: SoundSide, queueMode: Int): Boolean {
+        textToSpeak = tag.fieldText
+        mQuestionAnswer = qa
+        mDid = did
+        mOrd = ord
+        Timber.d("ReadText.textToSpeech() method started for string '%s', locale '%s'", tag.fieldText, tag.lang)
+        var localeCode = tag.lang
+        val originalLocaleCode = localeCode
+        if (!localeCode.isEmpty()) {
+            if (!isLanguageAvailable(localeCode)) {
+                localeCode = ""
             }
         }
-
-        /**
-         * Read the given text using an appropriate TTS voice.
-         *
-         *
-         * The voice is chosen as follows:
-         *
-         *
-         * 1. If localeCode is a non-empty string representing a locale in the format returned
-         * by Locale.toString(), and a voice matching the language of this locale (and ideally,
-         * but not necessarily, also the country and variant of the locale) is available, then this
-         * voice is used.
-         * 2. Otherwise, if the database contains a saved language for the given 'did', 'ord' and 'qa'
-         * arguments, and a TTS voice matching that language is available, then this voice is used
-         * (unless the saved language is NO_TTS, in which case the text is not read at all).
-         * 3. Otherwise, the user is asked to select a language from among those for which a voice is
-         * available.
-         *
-         * @param queueMode TextToSpeech.QUEUE_ADD or TextToSpeech.QUEUE_FLUSH.
-         * @return false if a sound was not played
-         */
-        private fun textToSpeech(tag: TTSTag, did: Long, ord: Int, qa: SoundSide, queueMode: Int): Boolean {
-            textToSpeak = tag.fieldText
-            mQuestionAnswer = qa
-            mDid = did
-            mOrd = ord
-            Timber.d("ReadText.textToSpeech() method started for string '%s', locale '%s'", tag.fieldText, tag.lang)
-            var localeCode = tag.lang
-            val originalLocaleCode = localeCode
-            if (!localeCode.isEmpty()) {
-                if (!isLanguageAvailable(localeCode)) {
-                    localeCode = ""
-                }
-            }
-            if (localeCode.isEmpty()) {
-                // get the user's existing language preference
-                localeCode = getLanguage(mDid, mOrd, mQuestionAnswer)
-                Timber.d("ReadText.textToSpeech() method found language choice '%s'", localeCode)
-            }
-            if (localeCode == NO_TTS) {
-                // user has chosen not to read the text
-                return false
-            }
-            if (!localeCode.isEmpty() && isLanguageAvailable(localeCode)) {
-                speak(textToSpeak, localeCode, queueMode)
-                return true
-            }
-
-            // Otherwise ask the user what language they want to use
-            if (!originalLocaleCode.isEmpty()) {
-                // (after notifying them first that no TTS voice was found for the locale
-                // they originally requested)
-                showThemedToast(
-                    mReviewer!!.get(),
-                    mReviewer!!.get()!!.getString(R.string.no_tts_available_message) +
-                        " (" + originalLocaleCode + ")",
-                    false
-                )
-            }
-            selectTts(textToSpeak, mDid, mOrd, mQuestionAnswer)
+        if (localeCode.isEmpty()) {
+            // get the user's existing language preference
+            localeCode = getLanguage(mDid, mOrd, mQuestionAnswer)
+            Timber.d("ReadText.textToSpeech() method found language choice '%s'", localeCode)
+        }
+        if (localeCode == NO_TTS) {
+            // user has chosen not to read the text
+            return false
+        }
+        if (!localeCode.isEmpty() && isLanguageAvailable(localeCode)) {
+            speak(textToSpeak, localeCode, queueMode)
             return true
         }
 
-        /**
-         * Returns true if the TTS engine supports the language of the locale represented by localeCode
-         * (which should be in the format returned by Locale.toString()), false otherwise.
-         */
-        private fun isLanguageAvailable(localeCode: String): Boolean {
-            return textToSpeech!!.isLanguageAvailable(LanguageUtils.localeFromStringIgnoringScriptAndExtensions(localeCode)) >=
-                TextToSpeech.LANG_AVAILABLE
+        // Otherwise ask the user what language they want to use
+        if (!originalLocaleCode.isEmpty()) {
+            // (after notifying them first that no TTS voice was found for the locale
+            // they originally requested)
+            showThemedToast(
+                mReviewer!!.get(),
+                mReviewer!!.get()!!.getString(R.string.no_tts_available_message) +
+                    " (" + originalLocaleCode + ")",
+                false
+            )
         }
+        selectTts(textToSpeak, mDid, mOrd, mQuestionAnswer)
+        return true
+    }
 
-        @JvmStatic
-        fun initializeTts(context: Context, listener: ReadTextListener) {
-            // Store weak reference to Activity to prevent memory leak
-            mReviewer = WeakReference(context)
-            mCompletionListener = listener
-            // Create new TTS object and setup its onInit Listener
-            textToSpeech = TextToSpeech(context) { status: Int ->
-                if (status == TextToSpeech.SUCCESS) {
-                    // build list of available languages
-                    buildAvailableLanguages()
-                    if (!availableTtsLocales.isEmpty()) {
-                        // notify the reviewer that TTS has been initialized
-                        Timber.d("TTS initialized and available languages found")
-                        (mReviewer!!.get() as AbstractFlashcardViewer?)!!.ttsInitialized()
-                    } else {
-                        showThemedToast(mReviewer!!.get(), mReviewer!!.get()!!.getString(R.string.no_tts_available_message), false)
-                        Timber.w("TTS initialized but no available languages found")
+    /**
+     * Returns true if the TTS engine supports the language of the locale represented by localeCode
+     * (which should be in the format returned by Locale.toString()), false otherwise.
+     */
+    private fun isLanguageAvailable(localeCode: String): Boolean {
+        return textToSpeech!!.isLanguageAvailable(LanguageUtils.localeFromStringIgnoringScriptAndExtensions(localeCode)) >=
+            TextToSpeech.LANG_AVAILABLE
+    }
+
+    @JvmStatic
+    fun initializeTts(context: Context, listener: ReadTextListener) {
+        // Store weak reference to Activity to prevent memory leak
+        mReviewer = WeakReference(context)
+        mCompletionListener = listener
+        // Create new TTS object and setup its onInit Listener
+        textToSpeech = TextToSpeech(context) { status: Int ->
+            if (status == TextToSpeech.SUCCESS) {
+                // build list of available languages
+                buildAvailableLanguages()
+                if (!availableTtsLocales.isEmpty()) {
+                    // notify the reviewer that TTS has been initialized
+                    Timber.d("TTS initialized and available languages found")
+                    (mReviewer!!.get() as AbstractFlashcardViewer?)!!.ttsInitialized()
+                } else {
+                    showThemedToast(mReviewer!!.get(), mReviewer!!.get()!!.getString(R.string.no_tts_available_message), false)
+                    Timber.w("TTS initialized but no available languages found")
+                }
+                textToSpeech!!.setOnUtteranceProgressListener(object : UtteranceProgressListener() {
+                    override fun onDone(arg0: String) {
+                        listener.onDone(getmQuestionAnswer())
                     }
-                    textToSpeech!!.setOnUtteranceProgressListener(object : UtteranceProgressListener() {
-                        override fun onDone(arg0: String) {
-                            listener.onDone(getmQuestionAnswer())
-                        }
 
-                        @Deprecated("")
-                        override fun onError(utteranceId: String) {
-                            Timber.v("Android TTS failed. Check logcat for error. Indicates a problem with Android TTS engine.")
-                            val helpUrl = Uri.parse(mReviewer!!.get()!!.getString(R.string.link_faq_tts))
-                            val ankiActivity = mReviewer!!.get() as AnkiActivity?
-                            ankiActivity!!.mayOpenUrl(helpUrl)
-                            showSnackbar(
-                                ankiActivity, R.string.no_tts_available_message, false, R.string.help,
-                                { openTtsHelpUrl(helpUrl) }, ankiActivity.findViewById(R.id.root_layout),
-                                    Snackbar.Callback()
-                                )
-                            }
-
-                            override fun onStart(arg0: String) {
-                                // no nothing
-                            }
-                        })
-                    } else {
-                        showThemedToast(mReviewer!!.get(), mReviewer!!.get()!!.getString(R.string.no_tts_available_message), false)
-                        Timber.w("TTS not successfully initialized")
+                    @Deprecated("")
+                    override fun onError(utteranceId: String) {
+                        Timber.v("Android TTS failed. Check logcat for error. Indicates a problem with Android TTS engine.")
+                        val helpUrl = Uri.parse(mReviewer!!.get()!!.getString(R.string.link_faq_tts))
+                        val ankiActivity = mReviewer!!.get() as AnkiActivity?
+                        ankiActivity!!.mayOpenUrl(helpUrl)
+                        showSnackbar(
+                            ankiActivity, R.string.no_tts_available_message, false, R.string.help,
+                            { openTtsHelpUrl(helpUrl) }, ankiActivity.findViewById(R.id.root_layout),
+                            Snackbar.Callback()
+                        )
                     }
-                }
-                // Show toast that it's getting initialized, as it can take a while before the sound plays the first time
-                showThemedToast(context, context.getString(R.string.initializing_tts), false)
-            }
 
-            private fun openTtsHelpUrl(helpUrl: Uri) {
-                val activity = mReviewer!!.get() as AnkiActivity?
-                activity!!.openUrl(helpUrl)
-            }
-
-            fun buildAvailableLanguages() {
-                availableTtsLocales.clear()
-                val systemLocales = Locale.getAvailableLocales()
-                availableTtsLocales.ensureCapacity(systemLocales.size)
-                for (loc in systemLocales) {
-                    try {
-                        val retCode = textToSpeech!!.isLanguageAvailable(loc)
-                        if (retCode >= TextToSpeech.LANG_COUNTRY_AVAILABLE) {
-                            availableTtsLocales.add(loc)
-                        } else {
-                            Timber.v("ReadText.buildAvailableLanguages() :: %s  not available (error code %d)", loc.displayName, retCode)
-                        }
-                    } catch (e: IllegalArgumentException) {
-                        Timber.w(e, "Error checking if language %s available", loc.displayName)
+                    override fun onStart(arg0: String) {
+                        // no nothing
                     }
-                }
-            }
-
-            /**
-             * Request that TextToSpeech is stopped and shutdown after it it no longer being used
-             * by the context that initialized it.
-             * No-op if the current instance of TextToSpeech was initialized by another Context
-             * @param context The context used during [.initializeTts]
-             */
-            @JvmStatic
-            fun releaseTts(context: Context) {
-                if (textToSpeech != null && mReviewer!!.get() === context) {
-                    textToSpeech!!.stop()
-                    textToSpeech!!.shutdown()
-                }
-            }
-
-            @JvmStatic
-            fun stopTts() {
-                if (textToSpeech != null) {
-                    textToSpeech!!.stop()
-                }
-            }
-
-            @JvmStatic
-            fun closeForTests() {
-                if (textToSpeech != null) {
-                    textToSpeech!!.shutdown()
-                }
-                textToSpeech = null
-                MetaDB.close()
-                System.gc()
-            }
-
-            interface ReadTextListener {
-                fun onDone(playedSide: SoundSide?)
+                })
+            } else {
+                showThemedToast(mReviewer!!.get(), mReviewer!!.get()!!.getString(R.string.no_tts_available_message), false)
+                Timber.w("TTS not successfully initialized")
             }
         }
-        
+        // Show toast that it's getting initialized, as it can take a while before the sound plays the first time
+        showThemedToast(context, context.getString(R.string.initializing_tts), false)
+    }
+
+    private fun openTtsHelpUrl(helpUrl: Uri) {
+        val activity = mReviewer!!.get() as AnkiActivity?
+        activity!!.openUrl(helpUrl)
+    }
+
+    fun buildAvailableLanguages() {
+        availableTtsLocales.clear()
+        val systemLocales = Locale.getAvailableLocales()
+        availableTtsLocales.ensureCapacity(systemLocales.size)
+        for (loc in systemLocales) {
+            try {
+                val retCode = textToSpeech!!.isLanguageAvailable(loc)
+                if (retCode >= TextToSpeech.LANG_COUNTRY_AVAILABLE) {
+                    availableTtsLocales.add(loc)
+                } else {
+                    Timber.v("ReadText.buildAvailableLanguages() :: %s  not available (error code %d)", loc.displayName, retCode)
+                }
+            } catch (e: IllegalArgumentException) {
+                Timber.w(e, "Error checking if language %s available", loc.displayName)
+            }
+        }
+    }
+
+    /**
+     * Request that TextToSpeech is stopped and shutdown after it it no longer being used
+     * by the context that initialized it.
+     * No-op if the current instance of TextToSpeech was initialized by another Context
+     * @param context The context used during [.initializeTts]
+     */
+    @JvmStatic
+    fun releaseTts(context: Context) {
+        if (textToSpeech != null && mReviewer!!.get() === context) {
+            textToSpeech!!.stop()
+            textToSpeech!!.shutdown()
+        }
+    }
+
+    @JvmStatic
+    fun stopTts() {
+        if (textToSpeech != null) {
+            textToSpeech!!.stop()
+        }
+    }
+
+    @JvmStatic
+    fun closeForTests() {
+        if (textToSpeech != null) {
+            textToSpeech!!.shutdown()
+        }
+        textToSpeech = null
+        MetaDB.close()
+        System.gc()
+    }
+
+    interface ReadTextListener {
+        fun onDone(playedSide: SoundSide?)
+    }
+}

--- a/AnkiDroid/src/main/java/com/ichi2/anki/SharedDecksDownloadFragment.kt
+++ b/AnkiDroid/src/main/java/com/ichi2/anki/SharedDecksDownloadFragment.kt
@@ -134,7 +134,7 @@ class SharedDecksDownloadFragment : Fragment() {
      * the download progress checker.
      */
     private fun downloadFile(fileToBeDownloaded: DownloadFile) {
-        // Register broadcast receiver for download completion. 
+        // Register broadcast receiver for download completion.
         Timber.d("Registering broadcast receiver for download completion")
         activity?.registerReceiver(mOnComplete, IntentFilter(DownloadManager.ACTION_DOWNLOAD_COMPLETE))
 

--- a/AnkiDroid/src/main/java/com/ichi2/anki/dialogs/AsyncDialogFragment.kt
+++ b/AnkiDroid/src/main/java/com/ichi2/anki/dialogs/AsyncDialogFragment.kt
@@ -25,7 +25,7 @@ import timber.log.Timber
 abstract class AsyncDialogFragment : AnalyticsDialogFragment() {
     /* provide methods for text to show in notification bar when the DialogFragment
        can't be shown due to the host activity being in stopped state.
-       This can happen when the DialogFragment is shown from 
+       This can happen when the DialogFragment is shown from
        the onPostExecute() method of an AsyncTask */
     @KotlinCleanup("convert these back to properties")
     abstract fun getNotificationMessage(): String?

--- a/AnkiDroid/src/main/java/com/ichi2/anki/dialogs/DatabaseErrorDialog.kt
+++ b/AnkiDroid/src/main/java/com/ichi2/anki/dialogs/DatabaseErrorDialog.kt
@@ -217,7 +217,7 @@ class DatabaseErrorDialog : AsyncDialogFragment() {
                             ImportFileSelectionFragment.openImportFilePicker(activity as AnkiActivity)
                         }
                         .itemsCallbackSingleChoice(-1) {
-                            _: MaterialDialog?, _: View?, which: Int, _: CharSequence? ->
+                                _: MaterialDialog?, _: View?, which: Int, _: CharSequence? ->
                             if (mBackups[which].length() > 0) {
                                 // restore the backup if it's valid
                                 (activity as DeckPicker?)

--- a/AnkiDroid/src/main/java/com/ichi2/anki/reviewer/PeripheralKeymap.kt
+++ b/AnkiDroid/src/main/java/com/ichi2/anki/reviewer/PeripheralKeymap.kt
@@ -1,17 +1,17 @@
 /*
- Copyright (c) 2020 David Allison <davidallisongithub@gmail.com>                     
-                                                                                     
- This program is free software; you can redistribute it and/or modify it under       
- the terms of the GNU General Public License as published by the Free Software       
- Foundation; either version 3 of the License, or (at your option) any later          
- version.                                                                            
-                                                                                     
- This program is distributed in the hope that it will be useful, but WITHOUT ANY     
- WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A     
- PARTICULAR PURPOSE. See the GNU General Public License for more details.            
-                                                                                     
- You should have received a copy of the GNU General Public License along with        
- this program.  If not, see <http://www.gnu.org/licenses/>.                          
+ Copyright (c) 2020 David Allison <davidallisongithub@gmail.com>
+
+ This program is free software; you can redistribute it and/or modify it under
+ the terms of the GNU General Public License as published by the Free Software
+ Foundation; either version 3 of the License, or (at your option) any later
+ version.
+
+ This program is distributed in the hope that it will be useful, but WITHOUT ANY
+ WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+ PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License along with
+ this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
 package com.ichi2.anki.reviewer

--- a/AnkiDroid/src/main/java/com/ichi2/libanki/DecksV16.kt
+++ b/AnkiDroid/src/main/java/com/ichi2/libanki/DecksV16.kt
@@ -311,7 +311,7 @@ class DecksV16(private val col: Collection, private val decksBackend: DecksBacke
         return this.all_names_and_ids(
             skip_empty_default = !force_default, include_filtered = dyn
         ).map {
-            x ->
+                x ->
             x.name
         }.toMutableList()
     }
@@ -704,10 +704,10 @@ class DecksV16(private val col: Collection, private val decksBackend: DecksBacke
     fun child_ids(parent_name: str): Iterable<did> {
         val prefix = parent_name + "::"
         return all_names_and_ids().filter {
-            x ->
+                x ->
             x.name.startsWith(prefix)
         }.map {
-            d ->
+                d ->
             d.id
         }.toMutableList()
     }

--- a/AnkiDroid/src/main/java/com/ichi2/libanki/ModelsV16.kt
+++ b/AnkiDroid/src/main/java/com/ichi2/libanki/ModelsV16.kt
@@ -371,7 +371,7 @@ class ModelsV16(col: Collection, backend: BackendV1) : ModelManager(col) {
     /** Mapping of field name : (ord, field). */
     fun fieldMap(m: NoteType): Map<str, Tuple<int, Field>> {
         return m.flds.jsonObjectIterable().map {
-            f ->
+                f ->
             Pair(f.getString("name"), Pair(f.getLong("ord"), f))
         }.toMap()
     }
@@ -594,7 +594,7 @@ and notes.mid = ? and cards.ord = ?""",
                 }
                 flds = Array(flds.size) { "" }
                 newflds.forEach {
-                    (i, fld) ->
+                        (i, fld) ->
                     flds[i] = fld
                 }
                 val fldsAsString = joinFields(flds)

--- a/AnkiDroid/src/main/java/com/ichi2/libanki/backend/DecksBackend.kt
+++ b/AnkiDroid/src/main/java/com/ichi2/libanki/backend/DecksBackend.kt
@@ -147,7 +147,7 @@ class RustDroidDeckBackend(private val backend: BackendV1) : DecksBackend {
 
     override fun all_names_and_ids(skip_empty_default: Boolean, include_filtered: Boolean): List<DeckNameId> {
         return backend.getDeckNames(skip_empty_default, include_filtered).entriesList.map {
-            entry ->
+                entry ->
             DeckNameId(entry.name, entry.id)
         }
     }

--- a/AnkiDroid/src/main/java/com/ichi2/libanki/utils/PythonExtensions.kt
+++ b/AnkiDroid/src/main/java/com/ichi2/libanki/utils/PythonExtensions.kt
@@ -94,7 +94,7 @@ fun JSONArray.remove(jsonObject: JSONObject) {
 
 fun JSONArray.index(jsonObject: JSONObject): Optional<Int> {
     this.jsonObjectIterable().forEachIndexed {
-        i, value ->
+            i, value ->
         run {
             if (jsonObject == value) {
                 return Optional.of(i)

--- a/AnkiDroid/src/main/java/com/ichi2/utils/JSONException.kt
+++ b/AnkiDroid/src/main/java/com/ichi2/utils/JSONException.kt
@@ -1,34 +1,34 @@
-/*  
+/*
  *  Copyright (c) 2020 Arthur Milchior <arthur@milchior.fr>
- *  
- *  This file is free software: you may copy, redistribute and/or modify it  
- *  under the terms of the GNU General Public License as published by the  
- *  Free Software Foundation, either version 3 of the License, or (at your  
- *  option) any later version.  
- *  
- *  This file is distributed in the hope that it will be useful, but  
- *  WITHOUT ANY WARRANTY; without even the implied warranty of  
- *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU  
- *  General Public License for more details.  
- *  
- *  You should have received a copy of the GNU General Public License  
+ *
+ *  This file is free software: you may copy, redistribute and/or modify it
+ *  under the terms of the GNU General Public License as published by the
+ *  Free Software Foundation, either version 3 of the License, or (at your
+ *  option) any later version.
+ *
+ *  This file is distributed in the hope that it will be useful, but
+ *  WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *  General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
  *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
- *  
- *  This file incorporates work covered by the following copyright and  
- *  permission notice:  
- *  
+ *
+ *  This file incorporates work covered by the following copyright and
+ *  permission notice:
+ *
  *    Copyright (c) 2002 JSON.org
- *    
+ *
  *    Permission is hereby granted, free of charge, to any person obtaining a copy
  *    of this software and associated documentation files (the "Software"), to deal
  *    in the Software without restriction, including without limitation the rights
  *    to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
  *    copies of the Software, and to permit persons to whom the Software is
  *    furnished to do so, subject to the following conditions:
- *   
+ *
  *    The above copyright notice and this permission notice shall be included in all
  *    copies or substantial portions of the Software.
- *   
+ *
  *    The Software shall be used for Good, not Evil.
  *
  *    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
@@ -37,7 +37,7 @@
  *    AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  *    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
  *    OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
- *    SOFTWARE. 
+ *    SOFTWARE.
  */
 
 package com.ichi2.utils

--- a/AnkiDroid/src/main/java/com/ichi2/utils/JSONTokener.kt
+++ b/AnkiDroid/src/main/java/com/ichi2/utils/JSONTokener.kt
@@ -1,34 +1,34 @@
-/*  
+/*
  *  Copyright (c) 2020 Arthur Milchior <arthur@milchior.fr>
- *  
- *  This file is free software: you may copy, redistribute and/or modify it  
- *  under the terms of the GNU General Public License as published by the  
- *  Free Software Foundation, either version 3 of the License, or (at your  
- *  option) any later version.  
- *  
- *  This file is distributed in the hope that it will be useful, but  
- *  WITHOUT ANY WARRANTY; without even the implied warranty of  
- *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU  
- *  General Public License for more details.  
- *  
- *  You should have received a copy of the GNU General Public License  
+ *
+ *  This file is free software: you may copy, redistribute and/or modify it
+ *  under the terms of the GNU General Public License as published by the
+ *  Free Software Foundation, either version 3 of the License, or (at your
+ *  option) any later version.
+ *
+ *  This file is distributed in the hope that it will be useful, but
+ *  WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *  General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
  *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
- *  
- *  This file incorporates work covered by the following copyright and  
- *  permission notice:  
- *  
+ *
+ *  This file incorporates work covered by the following copyright and
+ *  permission notice:
+ *
  *    Copyright (c) 2002 JSON.org
- *    
+ *
  *    Permission is hereby granted, free of charge, to any person obtaining a copy
  *    of this software and associated documentation files (the "Software"), to deal
  *    in the Software without restriction, including without limitation the rights
  *    to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
  *    copies of the Software, and to permit persons to whom the Software is
  *    furnished to do so, subject to the following conditions:
- *   
+ *
  *    The above copyright notice and this permission notice shall be included in all
  *    copies or substantial portions of the Software.
- *   
+ *
  *    The Software shall be used for Good, not Evil.
  *
  *    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
@@ -37,7 +37,7 @@
  *    AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  *    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
  *    OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
- *    SOFTWARE. 
+ *    SOFTWARE.
  */
 package com.ichi2.utils
 

--- a/lint-rules/build.gradle
+++ b/lint-rules/build.gradle
@@ -7,6 +7,7 @@ java {
 }
 
 ktlint {
+    version = "0.45.1"
     disabledRules = ["no-wildcard-imports"]
 }
 

